python/py-hgexperimental/Portfile

```diff
@@ -30,6 +31,13 @@ if {${name} ne ${subport}} {
                         port:py27-cython \
                         port:lz4
 
+    pre-configure {
+        # fix error: 'register' storage class specifier is deprecated and incompatible with C++17
+        # we build with -std=c++0x and 'register' is only deprecated
+        # see https://stackoverflow.com/questions/49692794/c-17-compatability-with-python-2-7
+        reinplace "s|else \"-Werror\"|else \"-Werror -Wno-error=deprecated-register\"|g" ${worksrcpath}/setup.py
```

Comment:
... what we usually use `reinplace` for is to substitute some variables. In this case it's way more robust (and kind of desired) to put the patch in a file and add `pathfiles` directive. What may happen with this kind of patching is that upstream changes just enough that the patch no longer applies, but you never notice the problem. If you use a proper patch, then the patch fails and gives you a signal that upstream changed something.

Reply:
good idea